### PR TITLE
Fix geolocation subdivision pattern matching

### LINKED ISSUE #2033: Regions are not being collected since July 14th
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Plausible Cloud from plausible.io

### Describe the bug

"I noticed when filtering by “Today” the regions don’t show. Is there a reason for that? Going to Map, Countries, and Cities shows data. If I go to “Last 7 days” it shows Regions."

[via twitter](https://twitter.com/ThisIsSingleton/status/1549137152906145792)

### Expected behavior

regions should show

### Screenshots

_No response_

### Environment

```markdown
- OS:
- Browser:
- Browser Version:
```

### PR DESCRIPTION
**Pull request by vinibrsl**
This commit fixes a bug where regions were not being saved. This was
caused because Geolix response was returning an additional
`:geolocation` map key. It also adds a test case for this.

Closes #2033

### Tests
- [X] Automated tests have been added

### Changelog
- [X] Entry has been added to changelog

### Documentation
- [X] This change does not need a documentation update

### Dark mode
- [X] This PR does not change the UI

### CODE REVIEW

**Comment by metmarkosaric** [comment]
thank you @vinibrsl!

File: lib/mix/tasks/download_country_database.ex [hunk 1]
```diff
@@ -1,4 +1,9 @@
 defmodule Mix.Tasks.DownloadCountryDatabase do
+  @moduledoc """
```
Comment (ukutaht) [inline lib/mix/tasks/download_country_database.ex:2]: Love these! :)

File: test/plausible_web/controllers/api/external_controller_test.exs [hunk 2]
```diff
@@ -636,20 +636,23 @@ defmodule PlausibleWeb.Api.ExternalControllerTest do
...
+    test "looks up location data from the ip address", %{conn: conn} do
       params = %{
         name: "pageview",
         domain: "external-controller-test-20.com",
         url: "http://gigride.live/"
       }
 
       conn
-      |> put_req_header("x-forwarded-for", "1.1.1.1")
+      |> put_req_header("x-forwarded-for", "2.2.2.2")
```
Comment (ukutaht) [inline test/plausible_web/controllers/api/external_controller_test.exs:647]: Hmm maybe the fake entry for `{1, 1, 1, 1}` in the config can be removed now?

**Review by ukutaht — APPROVED** [review]

File: test/plausible_web/controllers/api/external_controller_test.exs (hunk 2 above)
Comment (ukutaht) [reply test/plausible_web/controllers/api/external_controller_test.exs:647]: Cool, thanks

**Pull request merged**